### PR TITLE
Have different methods for float and int

### PR DESCRIPTION
**Pull request by thomaspoignant**
See https://github.com/open-feature/spec/issues/115 for the context.
Discussion also on [slack](https://cloud-native.slack.com/archives/C0344AANLA1/p1658587751496079).

### CODE REVIEW

File: specification/flag-evaluation.md
Comment (toddbaert) [inline specification/flag-evaluation.md:124]: `Conditional requirements` should be children of `Conditions`, so I think you want to put your "if" clause in a condition statement, and separate it from the requirement.

File: specification/flag-evaluation.md
Comment (toddbaert) [inline specification/flag-evaluation.md:108]: I think we should still include the parameters list, we can just merge it with the "typed flag evaluation" requirement.

**Review by toddbaert — CHANGES_REQUESTED** [review]
I agree fully with the concept, thanks for noticing! I left some comments on structure, and I don't think we want to lose the description of the params for the flag evaluation methods; we can just merge the typed/untyped requirements, while keeping the params:

>with parameters `flag key` (string, required), `default value` (boolean | number | string | structure, required), `evaluation context` (optional), and `evaluation options` (optional),

Added a few more SDK devs to the PR as well.

**Comment by toddbaert** [comment]
@justinabrahms @agentgonzo @InTheCloudDan pinging you all on this one, since it's relevant to Java, and also the Cloudbees SDK differentiates between floats/ints, while the LD SDK uses an abstraction that can be interpreted as either. Justin [suggested](https://github.com/open-feature/java-sdk/issues/32) we use something more generic, like [Number](https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html), which I'm also OK with, but I think in that case we still might want to add something to the spec as @thomaspoignant suggests to make sure we have some way of support both floating point and values.

I think personally, I would prefer to just provide separate methods as @thomaspoignant has suggested here, but I'm interested in other takes...

**Comment by justinabrahms** [comment]
I'm not opposed to separating floats vs ints. I'd just like to avoid it if we could.   But I'm hearing that maybe we can't? So... let's not. :)

File: specification/flag-evaluation.md
Comment (thomaspoignant) [reply specification/flag-evaluation.md]: I have fixed that.

I have also moved the `Conditional Requirement 1.3.3` to a **requirement** since it does not have any condition associated.

File: specification/flag-evaluation.md
Comment (thomaspoignant) [reply specification/flag-evaluation.md:108]: I have merged it.

**Comment by benjiro** [comment]
This sounds fine to me, do we need to be more explicit about types so it's not left up to interpretation? Eg
float vs float64(double) vs float128(decimal)
int32 vs int64(long)

Thoughts? @toddbaert @thomaspoignant

**Comment by thomaspoignant** [comment]
> float vs float64(double) vs float128(decimal)
> int32 vs int64(long)

It is a bit too detailed from my point of view.
Typescript does not have a diff between int and float, in java flot is not standard but they are using double, etc …
I would avoid being precise here.

**Comment by toddbaert** [comment]
Agreed. We also mention this, in the `types.md` doc:

>A numeric value of unspecified type or size. Implementation languages may further differentiate between integers, floating point numbers, and other specific numeric types and provide functionality as idioms dictate.

https://github.com/open-feature/spec/blob/main/specification/types.md#number

File: specification.json
Comment (toddbaert) [inline specification.json:75]: I think we should link to the "types" doc here if we say "all types"... but it might be better to say "strings, numbers, booleans and structures"

**Review by toddbaert — APPROVED** [review]
Sounds like we're in agreement. LGTM assuming we clarify the language around "all types".

File: specification.json
Comment (thomaspoignant) [reply specification.json]: I have made this change, so and we have added a link to types for more details.

**Comment by thomaspoignant** [comment]
BTW the ci is failing on this link `https://www.w3.org/TR/qaframe-spec/` that is working fine from where am I.

File: specification.json
Comment (toddbaert) [inline specification.json:69]:
```suggestion
            "content": "The implementation language differentiates between floating-point numbers and integers.",
```

**Review by toddbaert — APPROVED** [review]

**Comment by toddbaert** [comment]
>BTW the ci is failing on this link https://www.w3.org/TR/qaframe-spec/ that is working fine from where am I.

Re-ran it, seems to work now. Probably the site was slow for a time.

**Review by beeme1mr — APPROVED** [review]
This looks good to me too. It's a bummer we need it but not every language can be as great as JavaScript! 😄

**Comment by toddbaert** [comment]
Anyone want to add a final approval for this? Seems like we have consensus.

File: specification/flag-evaluation.md
Comment (justinabrahms) [inline specification/flag-evaluation.md:120]: I'm thinking this line should only include numbers because the condition above says "assuming your language differentiates between floating and non floating numbers.."?

Comment (thomaspoignant) [reply specification/flag-evaluation.md]: Oh good catch I've updated it.

**Review by justinabrahms — APPROVED** [review]

**Pull request merged**
